lib/main.js

```diff
@@ -319,6 +319,69 @@ if (options.interactiveTests) {
+        '  CONFIGJSON=/mnt/boot/config.json\n' +
+        'elif [ -f "/mnt/conf/config.json" ]; then\n' +
+        '  CONFIGJSON=/mnt/conf/config.json\n' +
+        'fi\n' +
+        'if [ -z "$SUPERVISOR_TAG" ]; then\n' +
+        '  echo "Couldn\'t extract SUPERVISOR_TAG from the supervisor.conf"\n' +
+        'elif [ -z "$CONFIGJSON" ]; then\n' +
+        '  echo "Couldn\'t find config.json, cannot continue".\n' +
+        'else\n' +
+        '  APIKEY="$(jq -r \'.apiKey // .deviceApiKey\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  DEVICEID="$(jq -r \'.deviceId\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  API_ENDPOINT="$(jq -r \'.apiEndpoint\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  SLUG="$(jq -r \'.deviceType\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  echo "SUPERVISOR_TAG: $SUPERVISOR_TAG"\n' +
+        // eslint-disable-next-line no-template-curly-in-string
+        '  SUPERVISOR_ID=$(curl -s "${API_ENDPOINT}/v3/supervisor_release?\\$select=id,image_name&\\$filter=((device_type%20eq' +
+        // eslint-disable-next-line no-template-curly-in-string
+        '%20\'$SLUG\')%20and%20(supervisor_version%20eq%20\'$SUPERVISOR_TAG\'))&apikey=${APIKEY}" | jq -e -r \'.d[0].id\')\n' +
+        '  echo "Extracted supervisor ID: $SUPERVISOR_ID"\n' +
+        // eslint-disable-next-line no-template-curly-in-string
+        '  API_REPLY=$(curl -s "${API_ENDPOINT}/v2/device($DEVICEID)?apikey=$APIKEY" -X PATCH -H \'Content-Type: application/' +
+        'json;charset=UTF-8\' --data-binary "{\\"supervisor_release\\": \\"$SUPERVISOR_ID\\"}")\n' +
+        '  if [ "$API_REPLY" = "OK" ]; then\n' +
+        '    if update-resin-supervisor | grep "Supervisor configuration found from API." ; then\n' +
+        '      TEST_OK=yes\n' +
+        '    fi\n' +
+        '  else\n' +
+        '    echo "Couldn\'t set Supervisor release in the API"\n' +
+        '  fi\n' +
+        'fi\n' +
+        'if [ "$TEST_OK" = "yes" ]; then\n' +
+        '  echo "Test SUCCESSFUL."\n' +
+        'else\n' +
+        '  echo "Test FAILED."\n' +
+        'fi'
+      ],
+      assert: [
+        'Setting the supervisor should output the supervisor release ID and an OK for the set: ' +
+        '"Extracted supervisor ID: 968"',
+        'If you see "Extracted supervisor ID: null", then check https://api.resin.io/v4/supervisor_release whether ' +
+        'there\'s a supervisor version equal to SUPERVISOR_TAG released for the given device type. ' +
+        'If there isn\'t, ping @team in i/supervisor, and retry the test when the supervisor is released properly.',
+        'The update-resin-supervisor run should have Supervisor configuration found from API., this signals that the ' +
+        'script has gotten the supervisor ID and information correctly from the API.',
```

Comment:
I don't think we need to provide details about what the result means. Lets just ask the tester to check if the output is X, otherwise its a failure.